### DISCUSSION TITLE
Support for OPDS "catalogs" feed

### DISCUSSION THREAD

**Discussion by danielweck** [edrlab/thorium-reader#1766] [Ideas]
I've just updated the JSON Schema, and I realised that support for is missing in Thorium.
See: https://github.com/opds-community/drafts/pull/38

Test URLs:

* http://libraries.aldiko.com/home.json
* https://libraryregistry.librarysimplified.org/libraries

**Comment by danielweck** [comment]
https://libraryregistry.librarysimplified.org/libraries

=>

http://readium2.herokuapp.com/opds-v2-browse/https%3A%2F%2Flibraryregistry.librarysimplified.org%2Flibraries

**Comment by danielweck** [comment]
http://libraries.aldiko.com/home.json

=>

http://readium2.herokuapp.com/opds-v2-browse/http%3A%2F%2Flibraries.aldiko.com%2Fhome.json

...for example click on the "france" `__href__` link:

https://libraries.aldiko.com/countries/france.json

=>

http://readium2.herokuapp.com/opds-v2-browse/https%3A%2F%2Flibraries.aldiko.com%2Fcountries%2Ffrance.json

(note that the "Canada" link is also a navigation feed, which exposes a selection of catalogs feeds (one additional level of indirection)

**Comment by danielweck** [comment]
Related: https://github.com/edrlab/thorium-reader/issues/274#issuecomment-856954435

**Comment by panaC** [comment]
https://github.com/readium/r2-opds-js/blob/e410866da1483632c756d9f3a3f1f09bb1666bbd/src/opds/opds2/opds2.ts#L153

**Comment by panaC** [comment]
with : https://libraryregistry.librarysimplified.org/libraries

<img width="1120" alt="image" src="https://user-images.githubusercontent.com/3198865/134168446-8a727ecb-69cf-40e8-b756-c27950bbfd32.png">


@llemeurfr do you agree with that ? 
the feed contains an array of `catalogs`

**Comment by llemeurfr** [comment]
@panaC I think that what is displayed in your screenshot is of no use for end users. 

If we want to support an OPDS "catalogs" structure in the Catalogs screen of Thorium, the goal must be to simplify the choice of one or more OPDS feeds inside such a set of feeds. The issue here is that entering the URL of the OPDS "catalogs" in a text field is a nerd thing (we see that with the current requirement to enter the URL of an OPDS catalog in Thorium).  
A discussion on the best way to do this in https://github.com/opds-community/drafts/discussions/47#discussioncomment-1415288. 

Once the OPDS "catalogs" URL is known from Thorium, we'd like to see an "OPDS feed picker" opening and be able to navigate into the (hierarchical) set of OPDS feeds to select one or more feeds, automatically added to Thorium. Aldiko Next offers just that, via the "Add" button in their "catalogs" screen.

**Comment by danielweck** [comment]
We need to flesh out the details for this feature, especially as there seems to be only 2 OPDS examples in the field that implement the draft specification. Moving to discussions...
